Agents/Xamarin.Interactive/Collections/ImmutableArray.cs

```diff
@@ -0,0 +1,51 @@
+﻿//
+// Author:
+//   Sandy Armstrong <sandy@xamarin.com>
+//
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Xamarin.Interactive.Collections
+{
+    /// <summary>
+    /// An intentionally simple, immutable array. To be replaced via NuGet/ILRepack when
+    /// we can depend on System.Collections.Immutable or some other NuGet.
+    /// </summary>
+    sealed class ImmutableArray<T> : IReadOnlyList<T>
```

Comment:
Ugh. So what all was using this that you needed to move? I think ultimately I want to just depend on S.C.Immutable (but we'll have to ILRepack).

Reply:
`Message` stuff and `AgentConnection`. And I agree. Fortunately there's very little `Add`ing going on so I think we can just ignore it until we have time to spend on ILRepack.

Reply:
I may not have needed to move `AgentConnection`. And I might be able to get away with only moving interface stuff for `Message`.

Reply:
Well, as it stands, we do need at least `Message.cs`. It would be more refactoring to change that. Like adding message creation API to `IMessageService`.

Reply:
I undid the `AgentConnection` move, but am leaving `Message` stuff alone. Splitting it up seems like it would just be frustrating.